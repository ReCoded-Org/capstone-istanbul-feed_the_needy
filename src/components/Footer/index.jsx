import React from "react";

const Footer = () => {
  return (
    <div>
      Footer here! 
    </div>
);
};

export default Footer;
