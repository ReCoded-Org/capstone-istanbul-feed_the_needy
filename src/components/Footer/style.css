.footer {
  display: flex;
  height: 12.5rem;
  background-color: #fa9600;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
}

.footerBottom {
  background-color: #637381;
  color: white;
}

.footerLogo {
  height: min-content;
}

.footerLinks a {
  text-decoration: none;
  color: white;
}

.footerSummary {
  font-size: xx-small;
  width: min-content;
  color: white;
}

.footerLocation {
  width: min-content;
  margin-top: 1em;
  display: flex;
  flex-direction: column;
}

.footerPhone {
  width: min-content;
  margin-top: 1em;
  display: flex;
  flex-direction: column;
}

.footerContact {
  display: flex;
  width: max-content;
}

.footerContact div {
  margin-right: 1em;
  margin-top: 1em;
  display: flex;
  flex-direction: column;
  height: min-content;
}

.footerContact div span {
  color: white;
}

.footerSocial {
  transform: scale(2);
  display: flex;
  height: min-content;
  margin-top: 5rem;
  margin-left: 10rem;
}

.footerMiddle {
  height: min-content;
  margin-left: 3rem;
}

@media (min-width: 768px) and (max-width: 1100px) {
  .footerSocial {
    margin-left: 4rem;
  }
}

@media (min-width: 576px) and (max-width: 768px) {
  .footerSocial {
    transform: scale(1.5);
    bottom: -8rem;
    position: absolute;
    left: 18.5rem;
  }

  .footerMiddle {
    margin-left: 0;
  }
}

@media (max-width: 576px) {
  .footer {
    flex-direction: column;
    height: auto;
  }

  .footerSocial div {
    flex-direction: row;
  }

  .footerSocial {
    margin: 3rem 0 1rem 8rem;
    transform: scale(2);
    width: min-content;
  }

  .footerContact {
    margin-left: 5.5rem;
    flex-direction: column;
    width: min-content;
  }

  .footerContact div {
    flex-direction: row;
  }

  .footerSummary {
    font-size: small;
  }

  .footerLinks {
    margin-top: 1rem;
    margin-left: 3rem;
  }

  .footerMiddle {
    margin-left: 0;
  }
}
