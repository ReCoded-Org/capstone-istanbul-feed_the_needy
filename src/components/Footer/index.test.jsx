import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Footer from "./index";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

it("Footer renders correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <Footer />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
