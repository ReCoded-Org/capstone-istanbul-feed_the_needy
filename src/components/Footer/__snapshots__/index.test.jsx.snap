// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
Array [
  <div
    className="footer"
  >
    <div
      style={
        Object {
          "margin": "3rem 3rem 0 3rem",
          "width": "min-content",
        }
      }
    >
      <div
        className="footerLogo"
      >
        <div
          className="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            className="ant-space-item"
            style={
              Object {
                "marginRight": 16,
              }
            }
          >
            <svg>
              logo.svg
            </svg>
          </div>
          <div
            className="ant-space-item"
            style={Object {}}
          >
            <svg>
              foodOnCoupon.svg
            </svg>
          </div>
        </div>
      </div>
      <span
        className="ant-typography footerSummary"
        style={
          Object {
            "WebkitLineClamp": null,
          }
        }
      >
        footer.summary
      </span>
    </div>
    <div
      className="footerMiddle"
    >
      <div
        className="footerLinks"
        style={
          Object {
            "height": "min-content",
            "marginTop": "4rem",
          }
        }
      >
        <div
          className="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            className="ant-space-item"
            style={
              Object {
                "marginRight": 50,
              }
            }
          >
            <a
              href="/"
              onClick={[Function]}
            >
              footer.headers.header0
            </a>
          </div>
          <div
            className="ant-space-item"
            style={
              Object {
                "marginRight": 50,
              }
            }
          >
            <a
              href="/coupons"
              onClick={[Function]}
            >
              footer.headers.header1
            </a>
          </div>
          <div
            className="ant-space-item"
            style={Object {}}
          >
            <a
              href="/blog"
              onClick={[Function]}
            >
              footer.headers.header2
            </a>
          </div>
        </div>
      </div>
      <div
        className="footerContact"
      >
        <div>
          <svg>
            location.svg
          </svg>
          <span
            className="ant-typography"
            style={
              Object {
                "WebkitLineClamp": null,
              }
            }
          >
            footer.location
          </span>
        </div>
        <div>
          <svg>
            phone.svg
          </svg>
          <span
            className="ant-typography"
            style={
              Object {
                "WebkitLineClamp": null,
              }
            }
          >
            +229-995-4438
          </span>
        </div>
        <div>
          <svg>
            mail.svg
          </svg>
          <span
            className="ant-typography"
            style={
              Object {
                "WebkitLineClamp": null,
              }
            }
          >
            team@fonCoupon.com
          </span>
        </div>
      </div>
    </div>
    <div
      className="footerSocial"
    >
      <div
        className="ant-space ant-space-horizontal ant-space-align-center"
      >
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <span
            aria-label="twitter"
            className="anticon anticon-twitter"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="twitter"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M928 254.3c-30.6 13.2-63.9 22.7-98.2 26.4a170.1 170.1 0 0075-94 336.64 336.64 0 01-108.2 41.2A170.1 170.1 0 00672 174c-94.5 0-170.5 76.6-170.5 170.6 0 13.2 1.6 26.4 4.2 39.1-141.5-7.4-267.7-75-351.6-178.5a169.32 169.32 0 00-23.2 86.1c0 59.2 30.1 111.4 76 142.1a172 172 0 01-77.1-21.7v2.1c0 82.9 58.6 151.6 136.7 167.4a180.6 180.6 0 01-44.9 5.8c-11.1 0-21.6-1.1-32.2-2.6C211 652 273.9 701.1 348.8 702.7c-58.6 45.9-132 72.9-211.7 72.9-14.3 0-27.5-.5-41.2-2.1C171.5 822 261.2 850 357.8 850 671.4 850 843 590.2 843 364.7c0-7.4 0-14.8-.5-22.2 33.2-24.3 62.3-54.4 85.5-88.2z"
              />
            </svg>
          </span>
        </div>
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <span
            aria-label="facebook"
            className="anticon anticon-facebook"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="facebook"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
              />
            </svg>
          </span>
        </div>
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <span
            aria-label="instagram"
            className="anticon anticon-instagram"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="instagram"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 378.7c-73.4 0-133.3 59.9-133.3 133.3S438.6 645.3 512 645.3 645.3 585.4 645.3 512 585.4 378.7 512 378.7zM911.8 512c0-55.2.5-109.9-2.6-165-3.1-64-17.7-120.8-64.5-167.6-46.9-46.9-103.6-61.4-167.6-64.5-55.2-3.1-109.9-2.6-165-2.6-55.2 0-109.9-.5-165 2.6-64 3.1-120.8 17.7-167.6 64.5C132.6 226.3 118.1 283 115 347c-3.1 55.2-2.6 109.9-2.6 165s-.5 109.9 2.6 165c3.1 64 17.7 120.8 64.5 167.6 46.9 46.9 103.6 61.4 167.6 64.5 55.2 3.1 109.9 2.6 165 2.6 55.2 0 109.9.5 165-2.6 64-3.1 120.8-17.7 167.6-64.5 46.9-46.9 61.4-103.6 64.5-167.6 3.2-55.1 2.6-109.8 2.6-165zM512 717.1c-113.5 0-205.1-91.6-205.1-205.1S398.5 306.9 512 306.9 717.1 398.5 717.1 512 625.5 717.1 512 717.1zm213.5-370.7c-26.5 0-47.9-21.4-47.9-47.9s21.4-47.9 47.9-47.9 47.9 21.4 47.9 47.9a47.84 47.84 0 01-47.9 47.9z"
              />
            </svg>
          </span>
        </div>
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <span
            aria-label="linkedin"
            className="anticon anticon-linkedin"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="linkedin"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM349.3 793.7H230.6V411.9h118.7v381.8zm-59.3-434a68.8 68.8 0 1168.8-68.8c-.1 38-30.9 68.8-68.8 68.8zm503.7 434H675.1V608c0-44.3-.8-101.2-61.7-101.2-61.7 0-71.2 48.2-71.2 98v188.9H423.7V411.9h113.8v52.2h1.6c15.8-30 54.5-61.7 112.3-61.7 120.2 0 142.3 79.1 142.3 181.9v209.4z"
              />
            </svg>
          </span>
        </div>
        <div
          className="ant-space-item"
          style={Object {}}
        >
          <span
            aria-label="youtube"
            className="anticon anticon-youtube"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="youtube"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M941.3 296.1a112.3 112.3 0 00-79.2-79.3C792.2 198 512 198 512 198s-280.2 0-350.1 18.7A112.12 112.12 0 0082.7 296C64 366 64 512 64 512s0 146 18.7 215.9c10.3 38.6 40.7 69 79.2 79.3C231.8 826 512 826 512 826s280.2 0 350.1-18.8c38.6-10.3 68.9-40.7 79.2-79.3C960 658 960 512 960 512s0-146-18.7-215.9zM423 646V378l232 133-232 135z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="footerBottom"
  >
    <span
      className="ant-typography"
      style={
        Object {
          "WebkitLineClamp": null,
          "marginLeft": "2rem",
        }
      }
    >
      footer.rights
    </span>
  </div>,
]
`;
